app/controllers/forgot_passwords_controller.rb

```diff
@@ -0,0 +1,25 @@
+class ForgotPasswordsController < ApplicationController
+  
+  def new; end
+  
+  def create
+    user = User.find_by(email_address: params[:email_address])
+    
+    if user
+      user.generate_token
+      AppMailer.send_forgot_password(user).deliver
+      redirect_to forgot_password_confirmation_path
+    else
+      if params[:email_address].blank?
+        flash[:danger] = "Email cannot be blank. Please try again."
+      else
+        flash[:danger] = "There is no user with that email in the system"
```

Comment:
Okay, it did cross my mind as to whether I should provide too much/too little info in this case. I thought in this case it would be good to let the user know that the email was def on the way but I guess it also means that someone can quickly keep hacking away at an email address and know straight away whether it was successful or not.  I'll make the change!